lib/string.gi

```diff
@@ -1258,9 +1258,121 @@ InstallGlobalFunction(StringOfMemoryAmount, function(m)
     Append(s, units[shift+1]);
     return s;
 end);
-
-        
     
+InstallGlobalFunction(PrintToFormatted, function(stream, s, args...)
+    local pos, len, outstr, nextbrace, endbrace,
+          argcounter, var,
+          splitBracket, toprint;
+
+    splitBracket := function(string, startpos, endpos)
```

Comment:
Perhaps add a brief comment saying what `splitBracket` does? It seems to parse a replacement field starting and ending at the given position in the string, is that correct? So perhaps `parseReplacementField` would be a better name?

Also note that the `string` argument could be replace, and instead `s` be accessed, as it is available in this closure.